Null Pointer Dereference Crash on 3-way Merge

I'm experimenting with dolt, and tried the conflict resolution behavior. For my test setup I've created simple dolt repository with simple tables with a foreign key:

```
$ dolt schema show
role @ working
CREATE TABLE `role` (
  `name` varchar(200) NOT NULL,
  `description` text,
  PRIMARY KEY (`name`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

user @ working
CREATE TABLE `user` (
  `id` bigint unsigned NOT NULL AUTO_INCREMENT,
  `name` varchar(200) NOT NULL,
  `first_name` varchar(200) DEFAULT NULL,
  `role_name` varchar(200) DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `role_name` (`role_name`),
  CONSTRAINT `fl0v22us` FOREIGN KEY (`role_name`) REFERENCES `role` (`name`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```

I did now add conflicting rows into master and a separate branch:

```
merge-comflict:
mysql> insert into role values ('#4', "merge conflict branch");
Query OK, 1 row affected (0.03 sec)

mysql> insert into user values (5, "Conflict", "Merge J.R.", "#4");
Query OK, 1 row affected (0.02 sec)

and master:
mysql> insert into role values ('#4', "master branch");
Query OK, 1 row affected (0.01 sec)

mysql> insert into user values (5, "Conflict", "Merge J.R. Master", "#4");
Query OK, 1 row affected (0.01 sec)
```

Trying to merge this togehter results in

```
$ dolt merge merge-conflict
Updating a1is08ne3bocfv6g92f3abnu3mv2fefk..jmk9i2vsun9o3o2p5k451v9dm8mlngqc
Auto-merging user
CONFLICT (content): Merge conflict in user
Auto-merging role
CONFLICT (content): Merge conflict in role
Automatic merge failed; fix conflicts and then commit the result.
$ dolt conflicts cat .
+-----+--------+----+----------+-------------------+-----------+
|     |        | id | name     | first_name        | role_name |
+-----+--------+----+----------+-------------------+-----------+
|  +  | ours   | 5  | Conflict | Merge J.R. Master | #4        |
|  +  | theirs | 5  | Conflict | Merge J.R.        | #4        |
+-----+--------+----+----------+-------------------+-----------+
+-----+--------+------+-----------------------+
|     |        | name | description           |
+-----+--------+------+-----------------------+
|  +  | ours   | #4   | master branch         |
|  +  | theirs | #4   | merge conflict branch |
+-----+--------+------+-----------------------+
$ dolt conflicts resolve role name
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c49c6e]

goroutine 1 [running]:
github.com/dolthub/dolt/go/libraries/doltcore/doltdb.(*Table).HashOf(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	github.com/dolthub/dolt/go/libraries/doltcore/doltdb/table.go:334 +0x4e
github.com/dolthub/dolt/go/cmd/dolt/commands/cnfcmds.manualResolve(0x28b6900, 0xc00016a2d0, 0xc00016b140, 0xc00022a000, 0xc000124400, 0xc00016b140)
	github.com/dolthub/dolt/go/cmd/dolt/commands/cnfcmds/resolve.go:201 +0xaba
github.com/dolthub/dolt/go/cmd/dolt/commands/cnfcmds.ResolveCmd.Exec(0x28b6900, 0xc00016a2d0, 0xc000696160, 0x16, 0xc0001b4030, 0x2, 0x2, 0xc00022a000, 0x16)
	github.com/dolthub/dolt/go/cmd/dolt/commands/cnfcmds/resolve.go:110 +0x2cc
github.com/dolthub/dolt/go/cmd/dolt/cli.SubCommandHandler.Exec(0x261ade7, 0x9, 0x2662d77, 0x33, 0xc000731460, 0x2, 0x2, 0x0, 0x28b6900, 0xc00016a2d0, ...)
	github.com/dolthub/dolt/go/cmd/dolt/cli/command.go:165 +0x4bd
github.com/dolthub/dolt/go/cmd/dolt/cli.SubCommandHandler.Exec(0x2613602, 0x4, 0x2628438, 0x11, 0xc0003e0480, 0x24, 0x24, 0x0, 0x28b6880, 0xc00019e000, ...)
	github.com/dolthub/dolt/go/cmd/dolt/cli/command.go:165 +0x4bd
main.runMain(0x0)
	github.com/dolthub/dolt/go/cmd/dolt/dolt.go:251 +0xec9
main.main()
	github.com/dolthub/dolt/go/cmd/dolt/dolt.go:110 +0x25
```

This is running dolt 0.22.11 via Homebrew on MacOS X Catalina 10.15.7 

````
$ brew info dolt
zsh: correct 'dolt' to '.dolt' [nyae]? n
dolt: stable 0.22.11 (bottled)
Git for Data
https://github.com/liquidata-inc/dolt
/usr/local/Cellar/dolt/0.22.11 (7 files, 56.3MB) *
  Poured from bottle on 2021-01-29 at 16:15:05
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/dolt.rb
License: Apache-2.0
==> Dependencies
Build: go ✘
```

--- a/go/cmd/dolt/commands/cnfcmds/resolve.go
+++ b/go/cmd/dolt/commands/cnfcmds/resolve.go
@@ -120,7 +120,7 @@ func autoResolve(ctx context.Context, apr *argparser.ArgParseResults, dEnv *env.
 	funcFlags := apr.FlagsEqualTo(autoResolverParams, true)
 
 	if funcFlags.Size() > 1 {
-		ff := strings.Join(autoResolverParams,  ", ")
+		ff := strings.Join(autoResolverParams, ", ")
 		return errhand.BuildDError("specify a resolver func from [ %s ]", ff).SetPrintUsage().Build()
 	} else if apr.NArg() == 0 {
 		return errhand.BuildDError("specify at least one table to resolve conflicts").SetPrintUsage().Build()

--- a/go/cmd/dolt/commands/cnfcmds/resolve.go
+++ b/go/cmd/dolt/commands/cnfcmds/resolve.go
@@ -16,6 +16,7 @@ package cnfcmds
 
 import (
 	"context"
+	"strings"
 
 	eventsapi "github.com/dolthub/dolt/go/gen/proto/dolt/services/eventsapi/v1alpha1"
 	"github.com/dolthub/dolt/go/libraries/doltcore/row"
@@ -119,9 +120,10 @@ func autoResolve(ctx context.Context, apr *argparser.ArgParseResults, dEnv *env.
 	funcFlags := apr.FlagsEqualTo(autoResolverParams, true)
 
 	if funcFlags.Size() > 1 {
-		return errhand.BuildDError("").SetPrintUsage().Build()
+		ff := strings.Join(autoResolverParams,  ", ")
+		return errhand.BuildDError("specify a resolver func from [ %s ]", ff).SetPrintUsage().Build()
 	} else if apr.NArg() == 0 {
-		return errhand.BuildDError("").SetPrintUsage().Build()
+		return errhand.BuildDError("specify at least one table to resolve conflicts").SetPrintUsage().Build()
 	}
 
 	autoResolveFlag := funcFlags.AsSlice()[0]
@@ -151,11 +153,10 @@ func manualResolve(ctx context.Context, apr *argparser.ArgParseResults, dEnv *en
 	args := apr.Args()
 
 	if len(args) < 2 {
-		return errhand.BuildDError("").SetPrintUsage().Build()
+		return errhand.BuildDError("at least two args are required").SetPrintUsage().Build()
 	}
 
 	root, verr := commands.GetWorkingWithVErr(dEnv)
-
 	if verr != nil {
 		return verr
 	}
@@ -169,49 +170,43 @@ func manualResolve(ctx context.Context, apr *argparser.ArgParseResults, dEnv *en
 	}
 
 	tbl, _, err := root.GetTable(ctx, tblName)
-
 	if err != nil {
 		return errhand.BuildDError("error: failed to get table '%s'", tblName).AddCause(err).Build()
 	}
 
 	sch, err := tbl.GetSchema(ctx)
-
 	if err != nil {
 		return errhand.BuildDError("error: failed to get schema").AddCause(err).Build()
 	}
 
 	keysToResolve, err := cli.ParseKeyValues(ctx, root.VRW(), sch, args[1:])
-
 	if err != nil {
 		return errhand.BuildDError("error: parsing command line").AddCause(err).Build()
 	}
+	if keysToResolve == nil {
+		return errhand.BuildDError("no primary keys were given to be resolved").Build()
+	}
 
 	invalid, notFound, updatedTbl, err := tbl.ResolveConflicts(ctx, keysToResolve)
-
 	if err != nil {
 		return errhand.BuildDError("fatal: Failed to resolve conflicts").AddCause(err).Build()
 	}
-
 	for _, key := range invalid {
 		cli.Printf("(%s) is not a valid key\n", row.TupleFmt(ctx, key.(types.Tuple)))
 	}
-
 	for _, key := range notFound {
 		cli.Printf("(%s) is not the primary key of a conflicting row\n", row.TupleFmt(ctx, key.(types.Tuple)))
 	}
-
 	if updatedTbl == nil {
 		return errhand.BuildDError("error: No changes were resolved").Build()
 	}
 
 	updatedHash, err := updatedTbl.HashOf()
-
 	if err != nil {
 		return errhand.BuildDError("error: failed to get table hash").AddCause(err).Build()
 	}
 
 	hash, err := tbl.HashOf()
-
 	if err != nil {
 		return errhand.BuildDError("error: failed to get table hash").AddCause(err).Build()
 	}

